optimum/onnxruntime/quantization.py

```diff
@@ -81,71 +84,91 @@ class ORTQuantizer(ABC):
     Handles the ONNX Runtime quantization process for models shared on huggingface.co/models.
     """
 
-    @staticmethod
+    @classmethod
     def from_pretrained(
-        model_name_or_path: Union[str, os.PathLike], feature: str, opset: Optional[int] = None
+        cls,
+        model_name_or_path: Union[str, os.PathLike],
+        use_auth_token: Optional[Union[bool, str, None]] = None,
+        from_transformers: Optional[bool] = False,
+        task: Optional[str] = None,
+        file_name: Optional[str] = None,
+        cache_dir: Optional[str] = None,
     ) -> "ORTQuantizer":
         """
         Instantiate a `ORTQuantizer` from a pretrained pytorch model and preprocessor.
 
         Args:
             model_name_or_path (`Union[str, os.PathLike]`):
                 Repository name in the Hugging Face Hub or path to a local directory hosting the model.
-            feature (`str`):
-                Feature to use when exporting the model.
-            opset (`int`, *optional*):
-                ONNX opset version to export the model with.
-
+            use_auth_token (`str` or `bool`, *optional*):
+                Is needed to load models from a private repository
+            from_transformers (`bool`, *optional*, defaults to `False`):
+                Defines wether the provided `model_name_or_path` contains a vanilla Transformers checkpoint.
+                ORTQuantizer will then convert the model first
+            task (`str`, *optional*):
+                Transformers pipeline task for the model. Will be used to convert the model if needed.
+            file_name(`str`, *optional*):
+                Overwrites the default model file name from `"model.onnx"` to `file_name`. This allows you to load different model files from the same
+                repository or directory.
+            cache_dir (`Union[str, Path]`, *optional*):
+                Path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
         Returns:
             An instance of `ORTQuantizer`.
         """
-        preprocessor = get_preprocessor(model_name_or_path)
-        model_class = FeaturesManager.get_model_class_for_feature(feature)
-        model = model_class.from_pretrained(model_name_or_path)
-
-        return ORTQuantizer(preprocessor, model, feature, opset)
+        model_file_name = ONNX_WEIGHTS_NAME if file_name is None else file_name
```

Comment:
This integration is not compatible with seq2seq models, I suggest we should integrate as well (and take their support into consideration) during the `ORTQuantizer` refactorization

Reply:
How would you do that? especially for static quantization? I would probably start without and add support for it later. Since you can use the `ORTQuantizer` with the `model_file_name` to quantize the `encoder` and then the `decoder`.

Reply:
For me it makes sense for `ORTQuantizer` to support dynamic quantization for seq2seq models (and if static quantization mode is activated, to simply warn the user that it's currently not supported for seq2seq models). We can definitely add dynamic quantization for seq2seq models in a future PR, but we should not introduce new arguments that will be incompatible with seq2seq support (such as `file_name`). Maybe we could simplify the number of supported cases in `from_pretrained` (supporting ORTModels as inputs makes definitely sense, but the other cases could be removed)

Reply:
I would rather try to add this in another PR than in this one. Since the Seq2Seq require a few more checks, e.g. if model files exists etc. also i am not fully sure how this. 
I can add a check if `feature` includes `seq2seq` or `ORTModelForSeq2Seq` to raise an error that it is not yet implemented.

Reply:
> Maybe we could simplify the number of supported cases in from_pretrained (supporting ORTModels as inputs makes definitely sense, but the other cases could be removed)

Meaning you would remove 
* vanilla transformers and any converting we do?

Reply:
What we could do for mutli-file models, e.g. `encoder_model.onnx`, `decoder_model.onnx` is support list for `file_name`. Meaning people can do `ORTQuantizer.from_pretrained("path/to/model/or/id",file_name=["model1.onnx","model2.onnx"])` or `ORTQuantizer.from_pretrained(seq2seq_model)`